### DISCUSSION TITLE
OSM-ID for poIs

### DISCUSSION THREAD

**Discussion by dspangenberg** [systemed/tilemaker#370] [Q&A] [answered]
Hi,

to get the OSM-ID for pois in my map, I added the following line to my process.lua

`obj:Attribute("osmid", obj:Id())`

```lua
function WritePOI(obj,class,subclass,rank)
	local layer = "poi"
	if rank>4 then layer="poi_detail" end
	obj:LayerAsCentroid(layer)
	SetNameAttributes(obj)
	obj:AttributeNumeric("rank", rank)
        obj:Attribute("osmid", obj:Id())
	obj:Attribute("class", class)
	obj:Attribute("subclass", subclass)
end
```

On the first view, it's looks good, because there is an osmid for my pois. But only for a part of my samples, I get actually results with the ID via Nominatim or Overpass.

Has anyone a clue why?

Thank you and best regards,
Danny

**Comment by systemed** [comment, marked as answer]
The id could refer to a node or a way - if you can't find it through the OSM API at `/node/12345`, have you tried looking at `/way/12345`?

**Comment by dspangenberg** [comment]
Yes, at least via overpass.

```
node(2327096849);
way(2327096849);
out;
```

**Comment by systemed** [comment]
So there's a POI at https://www.openstreetmap.org/node/2327096849 - is that not what you're expecting?

**Reply by dspangenberg** [reply]
Obviously my overpass request is wrong. I made now some samples over Nominatim and I get either a result for the way or the node. Thank you!
